go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -253,7 +253,7 @@ func DemystifyPending(status v1.PodStatus) (pluginsCore.PhaseInfo, error) {
 									&pluginsCore.TaskInfo{OccurredAt: &t},
 								), nil
 
-							case "CreateContainerConfigError":
+							case "CreateContainerConfigError", "InvalidImageName":
```

Comment:
Can we add a unit test to specifically test for this condition of the InvalidImageName and as per the desciption the task would keep running , So can we have the test simulate and after the fix the test succeeds with desired outcome

Reply:
sure

Reply:
Hi @pmahindrakar-oss 
How do I simulate this situation? We don't have integration tests in Flyteplugin.
I already have a unit test for it below to make sure that the task phase will be `PhasePermanentFailure`. is it enough?

Reply:
The unit test can just send a `status v1.PodStatus` object populated with `status.ContainerStatuses` which have 
`"CreateContainerConfigError", "InvalidImageName":` one of these values and the expectation would be to return 
`pluginsCore.PhaseInfoFailure(finalReason, finalMessage, &pluginsCore.TaskInfo{
									OccurredAt: &t,...`

I think that should be good enough and in the previous case before your fix it should be returning
`return pluginsCore.PhaseInfoSystemRetryableFailure(finalReason, finalMessage, &pluginsCore.TaskInfo{
									OccurredAt: &t,
								}), nil`